### DISCUSSION TITLE
WARNING: Conda package 'moose-dev' is currently at version

### DISCUSSION THREAD

**Discussion by rfryeSigma** [idaholab/moose#28139] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I have tried this on Linux and on an Arm MAC.
I create the conda moose env with $ conda create -n moose moose-dev=2024.07.03 .
I activate moose. I clone moose and go into test and start making the tests, but get a warning
```
$ make -j 8
WARNING: Conda package 'moose-dev' is currently at version '2024.07.03' and the required version is '2024.07.10'.
The correct version can be installed via:
    conda install moose-dev=2024.07.10
```

But there is no 2024.07.10 version of moose-dev available for installation.

On the MAC, the compilation succeeds, but several tests fail:
controls/web_server_control.control/bool ...................................................... FAILED (CRASH)
controls/web_server_control.control/real ...................................................... FAILED (CRASH)
controls/web_server_control.control/int ....................................................... FAILED (CRASH)
controls/web_server_control.control/string .................................................... FAILED (CRASH)
controls/web_server_control.control/vec_real .................................................. FAILED (CRASH)
controls/web_server_control.control/vec_int ................................................... FAILED (CRASH)
controls/web_server_control.control/vec_string ................................................ FAILED (CRASH)
controls/web_server_control.get_postprocessor ................................................. FAILED (CRASH)
controls/web_server_control.parallel_consistent .................................. [min_cpus=2] FAILED (CRASH)
controls/web_server_control.errors/set_controllable_no_exist ................. FAILED (EXPECTED ERROR MISSING)
controls/web_server_control.errors/postprocessor_no_exist .................... FAILED (EXPECTED ERROR MISSING)
controls/web_server_control.errors/set_controllable_unregistered_type ........ FAILED (EXPECTED ERROR MISSING)
controls/web_server_control.errors/set_controllable_bad_convert_json ......... FAILED (EXPECTED ERROR MISSING)
controls/web_server_control.errors/set_controllable_vector_non_array ......... FAILED (EXPECTED ERROR MISSING)
--------------------------------------------------------------------------------------------------------------
Ran 4359 tests in 277.5 seconds. Average test time 0.3 seconds, maximum test time 15.1 seconds.
4345 passed, 119 skipped, 0 pending, 14 FAILED

Here are the details on the first failure:

auxkernels/nodal_aux_var.init_test ........................................................................ OK
controls/web_server_control.control/bool: Working Directory: /Users/rogerfrye/Documents/Git/moose/test/tests/controls/web_server_control
controls/web_server_control.control/bool: Running command: RUNAPP_COMMAND="/Users/rogerfrye/Documents/Git/moose/test/moose_test-opt -i control_reporter.i Outputs/json/file_base=control_reporter_bool Reporters/test/bool_value=true --error --error-override --libtorch-device cpu" /Users/rogerfrye/Documents/Git/moose/test/tests/controls/web_server_control/control_reporter.py bool_value
controls/web_server_control.control/bool: MooseControl: Determined file socket /var/folders/1t/nqsy6f3n425f57zhk6v75c880000gn/T/moose_control_kfe630 for communication
controls/web_server_control.control/bool: MooseControl: Spawning MOOSE with command "['/Users/rogerfrye/Documents/Git/moose/test/moose_test-opt', '-i', 'control_reporter.i', 'Outputs/json/file_base=control_reporter_bool', 'Reporters/test/bool_value=true', '--error', '--error-override', '--libtorch-device', 'cpu', 'Controls/web_server/file_socket=/var/folders/1t/nqsy6f3n425f57zhk6v75c880000gn/T/moose_control_kfe630']"
controls/web_server_control.control/bool: MooseControl: Waiting for the webserver to start on "http+unix:///var/folders/1t/nqsy6f3n425f57zhk6v75c880000gn/T/moose_control_kfe630"
controls/web_server_control.control/bool: Traceback (most recent call last):
controls/web_server_control.control/bool:   File "/Users/rogerfrye/miniforge/envs/moose/lib/python3.10/encodings/idna.py", line 165, in encode
controls/web_server_control.control/bool:     raise UnicodeError("label too long")
controls/web_server_control.control/bool: UnicodeError: label too long
controls/web_server_control.control/bool: 
controls/web_server_control.control/bool: The above exception was the direct cause of the following exception:
controls/web_server_control.control/bool: 
controls/web_server_control.control/bool: Traceback (most recent call last):
controls/web_server_control.control/bool:   File "/Users/rogerfrye/Documents/Git/moose/test/tests/controls/web_server_control/control_reporter.py", line 59, in <module>
controls/web_server_control.control/bool:     base_controller('web_server', run_control)
controls/web_server_control.control/bool:   File "/Users/rogerfrye/Documents/Git/moose/test/tests/controls/web_server_control/base_controller.py", line 29, in base_controller
controls/web_server_control.control/bool:     control.initialize()
controls/web_server_control.control/bool:   File "/Users/rogerfrye/Documents/Git/moose/python/MooseControl/MooseControl.py", line 286, in initialize
controls/web_server_control.control/bool:     if self.isListening():
controls/web_server_control.control/bool:   File "/Users/rogerfrye/Documents/Git/moose/python/MooseControl/MooseControl.py", line 213, in isListening
controls/web_server_control.control/bool:     r = self._requests_get(f'{self._url}/check')
controls/web_server_control.control/bool:   File "/Users/rogerfrye/Documents/Git/moose/python/MooseControl/MooseControl.py", line 142, in _requests_get
controls/web_server_control.control/bool:     return self._requests_wrapper('get', *args, **kwargs)
controls/web_server_control.control/bool:   File "/Users/rogerfrye/Documents/Git/moose/python/MooseControl/MooseControl.py", line 138, in _requests_wrapper
controls/web_server_control.control/bool:     return getattr(accessor, function_name)(*args, **kwargs)
controls/web_server_control.control/bool:   File "/Users/rogerfrye/miniforge/envs/moose/lib/python3.10/site-packages/requests/sessions.py", line 602, in get
controls/web_server_control.control/bool:     return self.request("GET", url, **kwargs)
controls/web_server_control.control/bool:   File "/Users/rogerfrye/miniforge/envs/moose/lib/python3.10/site-packages/requests/sessions.py", line 579, in request
controls/web_server_control.control/bool:     settings = self.merge_environment_settings(
controls/web_server_control.control/bool:   File "/Users/rogerfrye/miniforge/envs/moose/lib/python3.10/site-packages/requests/sessions.py", line 760, in merge_environment_settings
controls/web_server_control.control/bool:     env_proxies = get_environ_proxies(url, no_proxy=no_proxy)
controls/web_server_control.control/bool:   File "/Users/rogerfrye/miniforge/envs/moose/lib/python3.10/site-packages/requests/utils.py", line 832, in get_environ_proxies
controls/web_server_control.control/bool:     if should_bypass_proxies(url, no_proxy=no_proxy):
controls/web_server_control.control/bool:   File "/Users/rogerfrye/miniforge/envs/moose/lib/python3.10/site-packages/requests/utils.py", line 816, in should_bypass_proxies
controls/web_server_control.control/bool:     bypass = proxy_bypass(parsed.hostname)
controls/web_server_control.control/bool:   File "/Users/rogerfrye/miniforge/envs/moose/lib/python3.10/urllib/request.py", line 2654, in proxy_bypass
controls/web_server_control.control/bool:     return proxy_bypass_macosx_sysconf(host)
controls/web_server_control.control/bool:   File "/Users/rogerfrye/miniforge/envs/moose/lib/python3.10/urllib/request.py", line 2631, in proxy_bypass_macosx_sysconf
controls/web_server_control.control/bool:     return _proxy_bypass_macosx_sysconf(host, proxy_settings)
controls/web_server_control.control/bool:   File "/Users/rogerfrye/miniforge/envs/moose/lib/python3.10/urllib/request.py", line 2599, in _proxy_bypass_macosx_sysconf
controls/web_server_control.control/bool:     hostIP = socket.gethostbyname(hostonly)
controls/web_server_control.control/bool: UnicodeError: encoding with 'idna' codec failed (UnicodeError: label too long)
controls/web_server_control.control/bool: 
controls/web_server_control.control/bool: 
controls/web_server_control.control/bool: Exit Code: 1
controls/web_server_control.control/bool: ################################################################################
controls/web_server_control.control/bool: Tester failed, reason: CRASH
controls/web_server_control.control/bool: 
controls/web_server_control.control/bool ...................................................... FAILED (CRASH)

**Comment by loganharbour** [comment]
This is related to https://github.com/idaholab/moose/discussions/27999 and will be addressed in the future in https://github.com/idaholab/moose/issues/28001.

It is a known issue and can be ignored for the time being.

> But there is no 2024.07.10 version of moose-dev available for installation.

This is because you are on the `next` branch of moose. While this is the default, it is the default because we accept PRs into it. If you checkout either the `devel` or `master` branch, you'll have a version that has a publicly available environment. A new environment change is currently making its way through the `next` branch and will not be released until it gets past `next`.

**Comment by rfryeSigma** [comment]
Thank you for the reply. I’m sorry I missed the previous issue.
I assume I can also ignore the warning at the beginning of the tests:

WARNING: Conda package 'moose-dev' is currently at version '2024.07.03' and the required version is '2024.07.10'.
The correct version can be installed via:
    conda install moose-dev=2024.07.10

**Reply by loganharbour** [reply]
> I assume I can also ignore the warning at the beginning of the tests:
>
> WARNING: Conda package 'moose-dev' is currently at version '2024.07.03' and the required version is '2024.07.10'.
> The correct version can be installed via:
>     conda install moose-dev=2024.07.10

Yes, that's correct. As I stated above, because you're on the `next` branch, you're on a non-release version. You can switch to current `devel` or `master` and you won't see this issue.

Alternatively, within a few hours once things work through our CI pipeline, that version will be available on conda.
